app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryimage/EntryImageActivity.java

```diff
@@ -186,8 +214,24 @@ public void showContestStatusScreen() {
         startActivity(ContestStatusActivity.makeIntent(this));
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    @Override
+    public void checkStoragePermissions() {
```

Comment:
We may need a function that returns true in pre-M devices. Try it in a simulator running Lollipop and see if it breaks here. If not you probably dont need to change this or the other 23 only function.